include/npcomp/Dialect/Basicpy/IR/BasicpyDialect.td

```diff
@@ -96,6 +96,36 @@ def Basicpy_UnknownType : DialectType<Basicpy_Dialect,
   }];
 }
 
+def Basicpy_ListType : DialectType<Basicpy_Dialect,
+    CPred<"$_self.isa<::mlir::NPCOMP::Basicpy::ListType>()">,
+    "List type"> {
+  let typeDescription = [{
+    A Python list type. In its generic form, there are no constraints on the
```

Comment:
"generic form" carries some baggage for me (kind of like "canonical form" -- implying something is done to put it in a concrete state). Maybe "In the general case"?

Reply:
Reworded.